draft-packages/button/KaizenDraft/Button/components/GenericButton.tsx

```diff
@@ -70,8 +70,17 @@ type WorkingUndefinedProps = {
 export type LabelProps = LabelPropsGeneric &
   (WorkingProps | WorkingUndefinedProps)
 
-export type IconButtonProps = GenericProps &
+type NonEmptyString<T extends string> = T extends "" ? never : T
+
+type IconOverrideProps<S extends string> = {
+  label: NonEmptyString<S>
+  icon: React.SVGAttributes<SVGSymbolElement>
+}
+
+export type IconButtonProps<S extends string> = GenericProps &
+  IconOverrideProps<S> &
   (WorkingProps | WorkingUndefinedProps)
+
```

Comment:
I suggest if these props are being defined for `IconButton`, then these sit in the file with the component instead of in here